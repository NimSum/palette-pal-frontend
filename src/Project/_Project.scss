.Project {
  border-bottom: 1px solid rgb(232, 232, 232);
}

.project-heading {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 25px 30px 5px 30px;
  color: rgb(100, 100, 120);
}

.project-heading-left {
  display: flex;
  align-items: center;

  h3 {
    font-weight: 600;
    font-size: 16px;
  }
}

.fa-trash {
  font-size: 16px;
  color: $btn-color;

  &:hover {
    color: darkred;
  }
}

.project-palettes {
  display: flex;
  flex-wrap: wrap;
  margin: 0 10px;
}

.add-palette-btn {
  border: 1px solid $btn-color;
  border-radius: 10px;
  color: $btn-color;
  height: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  &:hover {
    color: $btn-hover;
    border-color: $btn-hover;
    cursor: pointer;
  }

  .plus-txt {
    font-size: 90px;
    margin: -40px 0 -20px 0;
    font-weight: 300;
  }
}