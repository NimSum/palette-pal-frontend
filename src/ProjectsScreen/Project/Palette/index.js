import React from 'react';
import PaletteColor from './PaletteColor';

function Palette(props) {
	const paletteColors = [];

  for (let i = 1; i <= 5; i++) {
    const color = props.data[`color_${i}`];
    
    paletteColors.push(<PaletteColor
      color={color}
      palette_id={props.data.id}
      id={`color_${i}`}
      key={`color_${i}`}
      project_id={props.projectID}
      updatePaletteData={props.updatePaletteData}
      format={props.format}
    />);
	}

	return (
		<div className="Palette">
			<i
				className="fas fa-times-circle"
				onClick={() => {
					props.updatePaletteData({ ...props.data, project_id: props.projectID }, 'delete');
				}}
			/>
			{paletteColors}
			<p
				className="palette-title"
				contentEditable
				suppressContentEditableWarning
				onBlur={e =>
					props.updatePaletteData(
						{
							palette_name: e.target.textContent,
							id: props.data.id,
							project_id: props.projectID
						},
						'update'
					)}
				onKeyDown={e => {
					if (e.keyCode === 13 || (e.target.textContent.length >= 15 && e.keyCode !== 8)) {
						e.preventDefault();
						props.updatePaletteData(
							{
								palette_name: e.target.textContent,
								id: props.data.id,
								project_id: props.projectID
							},
							'update'
						);
					}
				}}>
				{props.data.name}
			</p>
		</div>
	);
}

export default Palette;
