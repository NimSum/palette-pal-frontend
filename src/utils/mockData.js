const mockData = {
  mockDetailedProjects: [
    {
      "project_name": "Project ONE",
      "project_id": 1,
      "palette_name": "Palette ONE",
      "palette_id": 1,
      "color_1": "#000000",
      "color_2": "#000000",
      "color_3": "#319547",
      "color_4": "#000000",
      "color_5": "#834267"
    },
    {
      "project_name": "Project TWO",
      "project_id": 2,
      "palette_name": "Palette TWO",
      "palette_id": 2,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267"
    },
    {
      "project_name": "Project ONE",
      "project_id": 2,
      "palette_name": "Palette THREE",
      "palette_id": 3,
      "color_1": "#123456",
      "color_2": "#654321",
      "color_3": "#245216",
      "color_4": "#784895",
      "color_5": "#235686"
    }
  ],
  mockProjects: [
    {
      "id": 1,
      "project_name": "Project ONE",
      "created_at": "2019-07-05T21:39:51.436Z",
      "updated_at": "2019-07-05T21:39:51.436Z"
    },
    {
      "id": 2,
      "project_name": "Project TWO",
      "created_at": "2019-07-05T21:40:51.436Z",
      "updated_at": "2019-07-05T21:40:51.436Z"
    }
  ],
  mockPalettes: [
    {
      "id": 1,
      "palette_name": "Test ONE",
      "project_id": 1,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267",
      "created_at": "2019-07-05T21:39:57.227Z",
      "updated_at": "2019-07-05T21:39:57.227Z"
    },
    {
        "id": 2,
        "palette_name": "Test TWO",
        "project_id": 2,
        "color_1": "#000000",
        "color_2": "#000000",
        "color_3": "#319547",
        "color_4": "#000000",
        "color_5": "#834267",
        "created_at": "2019-07-06T01:24:45.881Z",
        "updated_at": "2019-07-06T01:24:45.881Z"
    }
  ]
}

export default mockData;