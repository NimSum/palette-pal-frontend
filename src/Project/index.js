import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import Palette from '../Palette';

class Project extends Component {
	constructor(props) {
		super(props);

		this.state = {
			showMenu: false
		};
	}

	toggleMenu = () => {
		this.setState({ showMenu: !this.state.showMenu });
	};

	render() {
		const palettes = this.props.data.palettes.map(palette => <Palette data={palette} />);

		return (
			<article className="Project">
				<div className="project-heading">
					<div className="project-heading-left">
						<i className="fas fa-folder-open" />
						<h3 className="project-title">{this.props.data.name}</h3>
					</div>
					<i className="fas fa-ellipsis-h" onClick={this.toggleMenu} />
				</div>
        <div className="project-palettes">
          {palettes}
          <NavLink exact to="/" className="Palette add-palette-btn">
            <p className="plus-txt">+</p>
            <p className="add-palette-txt">
              Add New<br />Palette
            </p>
          </NavLink>
        </div>
			</article>
		);
	}
}

export default Project;
