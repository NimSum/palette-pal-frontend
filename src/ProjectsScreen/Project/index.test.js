import React from 'react';
import Project from './index';
import { shallow } from 'enzyme';
import mockData from '../../utils/mockData';

describe('Project', () => {
  let wrapper;
  const mockUpdatePaletteData = jest.fn();
  const mockUpdateProjectData = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Project
      data={mockData.mockCleanedProject}
      updateProjectData={mockUpdateProjectData}
      updatePaletteData={mockUpdatePaletteData}
    />);
  })
  
  it('should match component Project snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateProjectData when focus is left on the title field', () => {
    const mockEvent = { target: { textContent: 'test text' } }
    
    wrapper.find('.project-title').simulate('blur', mockEvent);

    expect(mockUpdateProjectData).toHaveBeenCalledWith({project_name: 'test text', id: 2}, 'update');
  })

  it('should call updateProjectData when the enter key is pressed on the title field', () => {
    const mockEvent = { target: { textContent: 'test text' }, keyCode: 13, preventDefault: jest.fn() }

    wrapper.find('.project-title').simulate('blur', mockEvent);

    expect(mockUpdateProjectData).toHaveBeenCalledWith({ project_name: 'test text', id: 2 }, 'update');
  })

  it('should call updateProjectData when the trash icon is clicked', () => {
    wrapper.find('.fa-trash').simulate('click');

    expect(mockUpdateProjectData).toHaveBeenCalledWith(mockData.mockCleanedProject, 'delete');
  })
})