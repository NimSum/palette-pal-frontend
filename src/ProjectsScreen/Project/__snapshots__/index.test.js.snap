// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project shoud match component Project snapshot 1`] = `ShallowWrapper {}`;
